adapters/facebook.go

```diff
@@ -156,32 +177,50 @@ func (a *FacebookAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder
 		// BANNER
 		fbReqB, err := a.MakeOpenRtbBidRequest(req, bidder, params.PlacementId, pbs.MEDIA_TYPE_BANNER, pubId, i)
 		if err == nil {
-			err = json.NewEncoder(&requests[reqIndex]).Encode(fbReqB)
+			requests[reqIndex] = &fbReqB
 			reqIndex = reqIndex + 1
-			if err != nil {
-				return nil, err
-			}
 		}
 
 		// VIDEO
 		fbReqV, err := a.MakeOpenRtbBidRequest(req, bidder, params.PlacementId, pbs.MEDIA_TYPE_VIDEO, pubId, i)
 		if err == nil {
-			err = json.NewEncoder(&requests[reqIndex]).Encode(fbReqV)
+			requests[reqIndex] = &fbReqV
 			reqIndex = reqIndex + 1
-			if err != nil {
-				return nil, err
-			}
+
 		}
 
 	}
+	return requests[:reqIndex], nil
+}
+
+func (a *FacebookAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pbs.PBSBidder) (pbs.PBSBidSlice, error) {
+	ortbRequests, e := a.GenerateRequestsForFacebook(req, bidder)
+
+	if e != nil {
+		return nil, e
+	}
+
+	requests := make([]bytes.Buffer, len(ortbRequests))
+	for i, ortbRequest := range ortbRequests {
+		e = json.NewEncoder(&requests[i]).Encode(ortbRequest)
+		if e != nil {
+			return nil, e
+		}
+	}
 
 	ch := make(chan callOneResult)
-	for i, _ := range bidder.AdUnits {
+
+	for i, _ := range requests {
 		go func(bidder *pbs.PBSBidder, reqJSON bytes.Buffer) {
-			result, err := a.callOne(ctx, req, reqJSON)
+			result, err := a.callOne(ctx, reqJSON)
 			result.Error = err
 			if result.bid != nil {
 				result.bid.BidderCode = bidder.BidderCode
+				unit := bidder.LookupAdUnit(result.bid.AdUnitCode)
+				if unit != nil {
+					result.bid.Width = unit.Sizes[0].W
```

Comment:
is this code here because FB doesn't return H/W in the bid response? Does the logic here solve this? 
https://github.com/prebid/prebid-server/blob/master/pbs_light.go#L398

Reply:
Yes the code is here because FB doesn't return H/W. Since CreativeMediaType is not sent back yet with creative, so this logic does not solve the problem. Can improve in a separate PR.